### DISCUSSION TITLE
Get requested fields in resolver

### DISCUSSION THREAD

**Discussion by bartenra** [mirumee/ariadne#641] [Q&A] [answered]
Hello!

I am wondering if there is a library that allows me to get the requested fields on the object of a current resolver. You'd need to parse info (`GraphQLResolveInfo`) for this. In the JS world there's this:

https://www.npmjs.com/package/graphql-parse-resolve-info (see Stack Overflow question [here](https://stackoverflow.com/questions/54984035/how-to-know-which-fields-were-requested-in-a-graphql-query/54988767#54988767)).

My use case is that I want to add directives to a certain object like so:

```graphql
type MyObject {
  id: ID! @public
  field: String! @public
  otherField: [SomeOtherObject!]!
}
```

And then check in a resolver if only public fields are being requested (those with `@public` directive), in which case the permissions check should be skipped.

**Comment by rafalp** [comment, marked as answer]
I'm not aware of such library existing for `graphql-core` (library Ariadne uses for GraphQL implementation).
